jinja2/runtime.py

```diff
@@ -422,29 +422,17 @@ class LoopContext(LoopContextBase):
 
     def __init__(self, iterable, undefined, recurse=None, depth0=0):
         LoopContextBase.__init__(self, undefined, recurse, depth0)
-        self._iterator = iter(iterable)
-
-        # try to get the length of the iterable early.  This must be done
-        # here because there are some broken iterators around where there
-        # __len__ is the number of iterations left (i'm looking at your
-        # listreverseiterator!).
         try:
+            # Getting length of iterable 
+            iterable = tuple(iterable)
```

Comment:
Do we really want to exhaust the iterable immediately on creation? Not sure if there are cases where this may be problematic.

cc @mitsuhiko

Reply:
But as I can see exhaust of the iterable was done in filters.do_reverse. Do you think that here need changes?